contracts/schemes/TokenTrade.sol

```diff
@@ -0,0 +1,179 @@
+pragma solidity ^0.6.10;
+// SPDX-License-Identifier: GPL-3.0
+
+import "../votingMachines/VotingMachineCallbacks.sol";
+
+
+/**
+ * @title A scheme for join in a dao.
+ * - A member can be proposed to join in by sending a min amount of fee.
+ * - A member can ask to quite (RageQuit) a dao on any time.
+ * - A member can donate to a dao.
+ */
+contract TokenTrade is VotingMachineCallbacks, ProposalExecuteInterface {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    event TokenTradeProposed(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        string _descriptionHash,
+        address indexed _beneficiary,
+        IERC20 _sendToken,
+        uint256 _sendTokenAmount,
+        IERC20 _receiveToken,
+        uint256 _receiveTokenAmount
+    );
+
+    event TokenTradeProposalExecuted(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        IERC20 _sendToken,
+        uint256 _sendTokenAmount,
+        IERC20 _receiveToken,
+        uint256 _receiveTokenAmount
+    );
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, int256 _decision);
+
+    struct Proposal {
+        address beneficiary;
+        IERC20 sendToken;
+        uint256 sendTokenAmount;
+        IERC20 receiveToken;
+        uint256 receiveTokenAmount;
+        bool exist;
```

Comment:
does the `exist` var is really needed  ?
does checking other filed like `beneficiary` or `sendToken` is not enough ?